### PR TITLE
Redux-server-rendering App Generated v28

### PR DESCRIPTION
**Pull request by robwise**

### CODE REVIEW

File: tester_app/client/app/bundles/HelloWorld/components/HelloWorldWidget.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import React, { PropTypes } from 'react';
+import Immutable from 'immutable';
+
+// Super simple example of the simplest possible React component
+export default class HelloWorldWidget extends React.Component {
+  static propTypes = {
+    // We prefix all property and variable names pointing to Immutable.js objects with '$$'.
+    // This allows us to immediately know we don't call $$helloWorldStore['someProperty'], but instead use
+    // the Immutable.js  `get` API for Immutable.Map
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/components/HelloWorldWidget.jsx:12]: double space before `get`

File: tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes } from 'react';
+import HelloWorldWidget from '../components/HelloWorldWidget';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import Immutable from 'immutable';
+import * as helloWorldActionCreators from '../actions/helloWorldActionCreators';
+
+function select(state) {
+  // Which part of the Redux global state does our component want to receive as props?
+  // Note the use of `$$`` to prefix the property name because the value is of type Immutable.js
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx:10]: extra backtick

File: tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    // This corresponds to the value used in function select above.
+    $$helloWorldStore: PropTypes.instanceOf(Immutable.Map).isRequired,
+  }
+
+  render() {
+    const { dispatch, $$helloWorldStore } = this.props;
+    const actions = bindActionCreators(helloWorldActionCreators, dispatch);
+    return (
+
+      // This uses the ES2015 spread operator to pass properties
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx:27]: move above line 25

Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx:27]: include comment so they don't need to see article

File: tester_app/client/app/bundles/HelloWorld/reducers/index.jsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// This file is our manifest of all reducers for the app.
+// See /client/app/bundles/HelloWorld/store/helloWorldStore.jsx
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/reducers/index.jsx:2]: add

// A real world app will like have many reducers and it helps to organize them in one file.
// See docs....

File: tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppClient.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import createStore from '../store/helloWorldStore';
+import HelloWorld from '../containers/HelloWorld';
+
+const helloWorldAppClient = props => {
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppClient.jsx:7]: // See documentation for https://github.com/rackt/react-redux.
// This is how you get props from the Rails view into the redux store.
// This code here binds your smart component to the redux store.

File: tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import createStore from '../store/helloWorldStore';
+import HelloWorld from '../containers/HelloWorld';
+
+const helloWorldAppServer = props => {
+  const store = createStore(props);
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx:8]: see comments above, and mention 

// This is how the server redux gets hydrated with data.

File: tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  const reactComponent = (
+    <Provider store={store}>
+      <HelloWorld />
+    </Provider>
+  );
+  return reactComponent;
+};
+
+// Export is needed for the hot reload server
```
Comment (justin808) [inline tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx:17]: remove comment

File: tester_app/client/app/lib/middlewares/loggerMiddleware.js [hunk 8]
```diff
@@ -0,0 +1,23 @@
+/* eslint no-console: 0 */
+import reduce from 'lodash/collection/reduce';
```
Comment (justin808) [inline tester_app/client/app/lib/middlewares/loggerMiddleware.js:2]: let's use regular lodash

File: tester_app/client/app/bundles/HelloWorld/components/HelloWorldWidget.jsx (hunk 1 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/components/HelloWorldWidget.jsx:12]: fixed

File: tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx (hunk 2 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx:10]: fixed

File: tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx (hunk 3 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/containers/HelloWorld.jsx:27]: done

File: tester_app/client/app/bundles/HelloWorld/reducers/index.jsx (hunk 4 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/reducers/index.jsx:2]: done

File: tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppClient.jsx (hunk 5 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppClient.jsx:7]: done

File: tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx (hunk 6 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx:8]: done

File: tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx (hunk 7 above)
Comment (robwise) [inline tester_app/client/app/bundles/HelloWorld/startup/helloWorldAppServer.jsx:17]: done

File: tester_app/client/app/lib/middlewares/loggerMiddleware.js (hunk 8 above)
Comment (robwise) [inline tester_app/client/app/lib/middlewares/loggerMiddleware.js:2]: done

**Pull request closed without merging**